src/main/java/com/icthh/xm/ms/entity/service/ExportImportService.java

```diff
@@ -0,0 +1,224 @@
+import com.icthh.xm.ms.entity.domain.XmEntity;
+import com.icthh.xm.ms.entity.repository.AttachmentRepository;
+import com.icthh.xm.ms.entity.repository.CalendarRepository;
+import com.icthh.xm.ms.entity.repository.EventRepository;
+import com.icthh.xm.ms.entity.repository.LinkRepository;
+import com.icthh.xm.ms.entity.repository.XmEntityRepository;
+import com.icthh.xm.ms.entity.service.dto.AttachmentExportDto;
+import com.icthh.xm.ms.entity.service.dto.CalendarExportDto;
+import com.icthh.xm.ms.entity.service.dto.EventExportDto;
+import com.icthh.xm.ms.entity.service.dto.LinkExportDto;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@Service
+@LepService(group = "service.exportImport")
+@Transactional
+@RequiredArgsConstructor
+public class ExportImportService {
+
+    private final XmEntityRepository entityRepository;
+    private final ObjectMapper objectMapper;
+    private final AttachmentRepository attachmentRepository;
+    private final CalendarRepository calendarRepository;
+    private final LinkRepository linkRepository;
+    private final EventRepository eventRepository;
+
+    @SneakyThrows
+    @LogicExtensionPoint("exportEntities")
+    public byte[] exportEntities(Set<ExportDto> exportEntities) {
+        ImportDto importDto = new ImportDto();
+        processXmEntities(exportEntities, importDto);
+
+        exportEntities.forEach(exportDto -> {
+            String typeKey = exportDto.getTypeKey();
+            processLinks(importDto, exportDto, typeKey);
+            processAttachments(importDto, exportDto, typeKey);
+            processCalendars(importDto, exportDto, typeKey);
+        });
+        return objectMapper.writeValueAsString(importDto).getBytes();
+    }
+
+    @LogicExtensionPoint("importEntities")
+    public void importEntities(ImportDto importDto) {
+        saveEntities(importDto);
+        saveLinks(importDto);
+        saveCalendars(importDto);
+        saveAttachments(importDto);
+    }
+
+    private void processXmEntities(Set<ExportDto> exportEntities, ImportDto importDto) {
+        Set<String> entityTypeKeys = exportEntities.stream().map(ExportDto::getTypeKey).collect(toSet());
+        List<XmEntity> entities = entityRepository.findAllByTypeKeyIn(of(0, MAX_VALUE), entityTypeKeys)
+                .getContent();
+        importDto.getEntities().addAll(entities);
+        log.info("Found export entities count: {}", entities.size());
+    }
+
+    private void processLinks(ImportDto importDto, ExportDto exportDto, String typeKey) {
+        List<String> linkTypeKeys = exportDto.getLinkTypeKeys();
+        List<Link> links = linkRepository.findBySourceTypeKeyAndTargetTypeKeyIn(typeKey, linkTypeKeys);
```

Comment:
describe why TargetTypeKeyIn maybe simple TypeKeyIn
how about case when we have 2 type of links from user to user like friends and family, and user want export only friends?